pyqg/model.py

```diff
@@ -233,6 +237,35 @@ def __init__(
         self._initialize_inversion_matrix()
         self._initialize_diagnostics(diagnostics_list)
 
+    def add_particles(self, initial_coords=None, order=1):
+        """Add particles to the simulation whose positions will be updated at
+        each timestep.
+
+        Parameters
+        ----------
+
+        initial_coords : tuple
+            An optional tuple of x,y coordinates for the initial positions of
+            particles at each fluid layer.
+        order : int
+            Order of interpolation.
+        """
+
+        if initial_coords is None:
+            n = self.nx // 8
```

Comment:
```suggestion
            n = int(self.nx * density)
```

or maybe something even better (non-discretized, using linspace)